app/drivers/display/il0323.c

```diff
@@ -321,95 +300,76 @@ static int il0323_controller_init(const struct device *dev) {
 #endif /* panel width */
 
     LOG_HEXDUMP_DBG(tmp, 1, "PSR");
-    if (il0323_write_cmd(driver, IL0323_CMD_PSR, tmp, 1)) {
+    if (il0323_write_cmd(cfg, IL0323_CMD_PSR, tmp, 1)) {
         return -EIO;
     }
 
     /* Set panel resolution */
     tmp[IL0323_TRES_HRES_IDX] = EPD_PANEL_WIDTH;
     tmp[IL0323_TRES_VRES_IDX] = EPD_PANEL_HEIGHT;
     LOG_HEXDUMP_DBG(tmp, IL0323_TRES_REG_LENGTH, "TRES");
-    if (il0323_write_cmd(driver, IL0323_CMD_TRES, tmp, IL0323_TRES_REG_LENGTH)) {
+    if (il0323_write_cmd(cfg, IL0323_CMD_TRES, tmp, IL0323_TRES_REG_LENGTH)) {
         return -EIO;
     }
 
     tmp[IL0323_CDI_CDI_IDX] = DT_INST_PROP(0, cdi);
     LOG_HEXDUMP_DBG(tmp, IL0323_CDI_REG_LENGTH, "CDI");
-    if (il0323_write_cmd(driver, IL0323_CMD_CDI, tmp, IL0323_CDI_REG_LENGTH)) {
+    if (il0323_write_cmd(cfg, IL0323_CMD_CDI, tmp, IL0323_CDI_REG_LENGTH)) {
         return -EIO;
     }
 
     tmp[0] = DT_INST_PROP(0, tcon);
-    if (il0323_write_cmd(driver, IL0323_CMD_TCON, tmp, 1)) {
+    if (il0323_write_cmd(cfg, IL0323_CMD_TCON, tmp, 1)) {
         return -EIO;
     }
 
     /* Enable Auto Sequence */
     tmp[0] = IL0323_AUTO_PON_DRF_POF;
-    if (il0323_write_cmd(driver, IL0323_CMD_AUTO, tmp, 1)) {
+    if (il0323_write_cmd(cfg, IL0323_CMD_AUTO, tmp, 1)) {
         return -EIO;
     }
 
     return 0;
 }
 
 static int il0323_init(const struct device *dev) {
-    struct il0323_data *driver = dev->data;
-
-    LOG_DBG("");
+    const struct il0323_cfg *cfg = dev->config;
 
-    driver->spi_dev = device_get_binding(IL0323_BUS_NAME);
-    if (driver->spi_dev == NULL) {
-        LOG_ERR("Could not get SPI device for IL0323");
+    if (!spi_is_ready_dt(&cfg->spi)) {
```

Comment:
There seems to be an issue with `spi_is_ready_dt` on Zephyr 3.2.0 and SDK 0.15.2. 

Even though `spi_is_ready_dt` is documented (https://docs.zephyrproject.org/apidoc/latest/group__spi__interface.html#ga37b4e5079ed18b70b0c5a260f4c36403), it doesn't seem to be defined (running `grep -r spi_is_ready_dt zephyr/include` returns nothing). Can we replace it with `spi_is_ready` (which exists) or the additional check for the CS gpio important here?

Reply:
Yeah, fixed that, I had a mixed up local environment pointing to Zephyr `main` while doing some build tests here. I believe in `main` it is renamed.